sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.cs

```diff
@@ -843,299 +729,336 @@ private string GetParmModifierForPointeeType(Decl decl, Type pointeeType)
+            {
+                name = GetTypeNameForPointeeType(namedDecl, referenceType.PointeeType);
+            }
+            else if (type is TypedefType typedefType)
+            {
+                if (!_config.GenerateUnsafeCode)
                 {
-                    return GetTypeName(decl, type.CanonicalType);
-                }
+                    name = GetTypeNameForUnderlyingType(namedDecl, typedefType.UnderlyingType);
 
-                case CXTypeKind.CXType_ConstantArray:
-                case CXTypeKind.CXType_IncompleteArray:
-                {
-                    return GetTypeName(decl, type.ElementType);
+                    if (string.IsNullOrWhiteSpace(name))
+                    {
+                        name = GetRemappedCursorName(typedefType.Decl);
+                    }
                 }
 
-                default:
+                switch (name)
                 {
-                    var name = type.Spelling;
-                    AddDiagnostic(DiagnosticLevel.Warning, $"Unsupported type: '{type.KindSpelling}'. Falling back '{name}'.", decl);
-                    return name;
-                }
-            }
-        }
+                    case "int8_t":
+                    {
+                        name = "sbyte";
+                        break;
+                    }
 
-        private string GetTypeNameForPointeeType(Decl decl, Type pointeeType)
-        {
-            switch (pointeeType.Kind)
-            {
-                case CXTypeKind.CXType_Void:
-                {
-                    if (_config.GenerateUnsafeCode)
+                    case "int16_t":
                     {
-                        return "void*";
+                        name = "short";
+                        break;
                     }
 
-                    _outputBuilder.AddUsingDirective("System");
-                    return "IntPtr";
-                }
+                    case "int32_t":
+                    {
+                        name = "int";
+                        break;
+                    }
 
-                case CXTypeKind.CXType_UChar:
-                case CXTypeKind.CXType_UShort:
-                case CXTypeKind.CXType_UInt:
-                case CXTypeKind.CXType_ULong:
-                case CXTypeKind.CXType_ULongLong:
-                case CXTypeKind.CXType_SChar:
-                case CXTypeKind.CXType_Short:
-                case CXTypeKind.CXType_Int:
-                case CXTypeKind.CXType_Long:
-                case CXTypeKind.CXType_LongLong:
-                case CXTypeKind.CXType_Float:
-                case CXTypeKind.CXType_Double:
-                case CXTypeKind.CXType_Pointer:
-                case CXTypeKind.CXType_Record:
-                case CXTypeKind.CXType_Enum:
-                case CXTypeKind.CXType_Typedef:
-                {
-                    switch (decl.Kind)
+                    case "int64_t":
+                    case "time_t":
```

Comment:
`time_t` is generally 64-bit on modern hardware (even 32-bit machines), but not always. Should probably provide a way to override this....